stdlib/public/runtime/SwiftObject.mm

```diff
@@ -584,11 +596,13 @@ static bool isBridgeObjectTaggedPointer(void *object) {
 
 #if SWIFT_OBJC_INTEROP
   if (!isNonNative_unTagged_bridgeObject(object)) {
```

Comment:
How expensive is this to be pushed down and simply merge the two paths into one?  AIUI, it should be a `csel` and `tbnz` right?  I suppose that you want to avoid the 2-instructions?

Reply:
The two paths have to be different, because we can't tail-call `objc_retain`. We have to mask off `objectPointerIsObjCBit` to make the call, but we have to return the pointer with `objectPointerIsObjCBit` set. However, `swift_retain` is OK because bridge objects pointing to Swift objects don't have any extra bits set, so the tail call works there.